schemas/mozilla_nimbus_schemas/experiments/experiments.py

```diff
@@ -1,84 +1,272 @@
+        )
+    )
+    isRollout: bool | SkipJsonSchema[None] = Field(
+        description=(
+            "When this property is set to true, treat this experiment as a rollout.\n"
+            "\n"
+            "Rollouts are currently handled as single-branch experiments separated from "
+            "the bucketing namespace for normal experiments.\n"
+            "\n"
+            "See-also: https://mozilla-hub.atlassian.net/browse/SDK-405"
+        ),
+        default=None,
+    )
+    bucketConfig: ExperimentBucketConfig = Field(description="Bucketing configuration.")
+    outcomes: list[ExperimentOutcome] | SkipJsonSchema[None] = Field(
+        description="A list of outcomes relevant to the experiment analysis.",
+        default=None,
+    )
+    featureIds: list[str] | SkipJsonSchema[None] = Field(
+        description="A list of featureIds the experiment contains configurations for.",
+        default=None,
+    )
+    branches: Union[
+        list[ExperimentSingleFeatureBranch],
+        list[ExperimentMultiFeatureDesktopBranch],
+        list[ExperimentMultiFeatureMobileBranch],
+    ] = Field(description="Branch configuration for the experiment.")
+    targeting: str | None = None
+    startDate: str | None = Field(
+        description=(
+            "Actual publish date of the experiment.\n"
+            "\n"
+            "Note that this value is expected to be null in Remote Settings."
+        ),
+    )
+    enrollmentEndDate: Optional[str] = Field(
+        description=(
+            "Actual enrollment end date of the experiment.\n"
+            "\n"
+            "Note that this value is expected to be null in Remote Settings."
+        ),
+        default=None,
+    )
+    endDate: str | None = Field(
+        description=(
+            "Actual end date of this experiment.\n"
+            "\n"
+            "Note that this field is expected to be null in Remote Settings."
+        ),
+    )
+    proposedDuration: int | SkipJsonSchema[None] = Field(
+        description=(
+            "Duration of the experiment from the start date in days.\n"
+            "\n"
+            "Note that this property is only used during the analysis phase (i.e., not "
+            "by the SDK)."
+        ),
+        default=None,
+    )
+    proposedEnrollment: int = Field(
+        description=(
+            "This represents the number of days that we expect to enroll new users.\n"
+            "\n"
+            "Note that this property is only used during the analysis phase (i.e., not "
+            "by the SDK)."
+        )
+    )
+    referenceBranch: str | None = Field(
+        description=(
+            'The slug of the reference branch (i.e., the branch we consider "control").'
+        )
+    )
+    featureValidationOptOut: bool | SkipJsonSchema[None] = Field(
+        description="Opt out of feature schema validation. Only supported on desktop.",
+        default=None,
+    )
+    localizations: dict[str, dict[str, str]] | None = Field(
```

Comment:
In theory we'd use like TypedDict here to actually specify the shape/keys, if they're known.

Reply:
The data looks like
```ts
{
  // ...
  localizations: {
    "en-US": { // limited to known locale codes that Firefox uses
      "foo": "foo-text", // these keys are arbitrary
      "bar": "bar-text"
    },
    "fr": {
      "foo": "foo-text en francais",
      "bar": "bar-text en francais",
    }
 }
}
```

so a more strongly typed version would be

```ts
type LocaleKey = "en-US" | "en-CA" | "fr" | ...;
type Localizations = Record<LocaleKey, Record<string, string>>;
```

but I dont think its worth enforcing the locale key correctness at the schema level since it is enforced in the client already